apps/www/components/Audio/AudioPlayer/ui/tabs/latest/LatestArticles.tsx

```diff
@@ -0,0 +1,161 @@
+    variables: {
+      count: 20,
+    },
+  })
+  const { addAudioQueueItem, checkIfInQueue, checkIfActiveItem } =
+    useAudioQueue()
+
+  const hasReadAloudDocuments =
+    data?.latestArticles?.nodes.some(
+      (node) => node?.meta?.audioSource?.kind === 'readAloud',
+    ) || false
+  if (hasReadAloudDocuments && filter !== 'all') {
+    setFilter('all')
+  }
+
+  const handlePlay = async (article: AudioPlayerItem) => {
+    await addAudioQueueItem(article, 1)
+    // TODO: handle error
+  }
+
+  const handleAddToQueue = async (article: AudioPlayerItem) => {
+    await addAudioQueueItem(article)
+    // TODO: handle error
+  }
+
+  const filteredArticles = useMemo(() => {
+    const articleWithAudio = data?.latestArticles?.nodes.filter(
+      (article) => !!article?.meta?.audioSource,
+    )
+
+    return articleWithAudio?.filter((article) => {
+      if (filter === 'all') {
+        return true
+      }
+      return article?.meta?.audioSource?.kind === 'readAloud'
+    })
+  }, [data, filter])
+
+  if (!hasAccess) {
+    return (
+      <NoAccess
+        text={t('AudioPlayer/Latest/NoAcces')}
+        heading={t('AudioPlayer/shared/NoAccess/heading')}
+      />
+    )
+  }
+
+  if (loading) {
+    return <LoadingPlaceholder />
+  }
+
+  if (error) {
+    return <div>Error: {error.message}</div>
+  }
+
+  return (
+    <div {...styles.root}>
+      {hasReadAloudDocuments && (
+        <div {...styles.filters}>
+          <FilterButton
+            isActive={filter === 'read-aloud'}
+            onClick={() => setFilter('read-aloud')}
+          >
+            {t('AudioPlayer/Latest/ReadAloud')}
+          </FilterButton>
+          <FilterButton
+            isActive={filter === 'all'}
+            onClick={() => setFilter('all')}
+          >
+            {t('AudioPlayer/Latest/All')}
+          </FilterButton>
+        </div>
+      )}
+      <ul {...styles.list}>
+        {filteredArticles.map((article) => (
+          <li key={article.id}>
+            <AudioListItem
+              item={article}
+              isActive={checkIfActiveItem(article.id)}
+              beforeActionItem={
+                <IconButton
+                  Icon={PlaylistAddIcon}
+                  title={t('AudioPlayer/Queue/Add')}
+                  onClick={() => handleAddToQueue(article)}
+                  disabled={checkIfInQueue(article.id)}
+                  style={{ marginRight: 0 }}
+                />
+              }
+              actions={[
+                {
+                  Icon: DownloadIcon,
+                  label: t('AudioPlayer/Queue/Download'),
+                  onClick: () => handleDownload(article),
+                },
+                {
+                  Icon: LinkIcon,
+                  label: t('AudioPlayer/Queue/GoToItem'),
+                  onClick: () => handleOpenArticle(article.meta.path),
+                },
+              ]}
+              onClick={() => handlePlay(article)}
+            />
+          </li>
```

Comment:
Extract into a separate component and show a loader after calling handleAddToQueue